cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/NamespaceExistenceVerifier.java

```diff
@@ -19,27 +19,35 @@
 import co.cask.cdap.common.NamespaceNotFoundException;
 import co.cask.cdap.common.NotFoundException;
 import co.cask.cdap.common.entity.EntityExistenceVerifier;
+import co.cask.cdap.common.namespace.NamespaceQueryAdmin;
 import co.cask.cdap.proto.id.NamespaceId;
-import co.cask.cdap.store.NamespaceStore;
+import com.google.common.base.Throwables;
 import com.google.inject.Inject;
 
 /**
  * {@link EntityExistenceVerifier} for {@link NamespaceId namespaces}.
  */
 public class NamespaceExistenceVerifier implements EntityExistenceVerifier<NamespaceId> {
 
-  private final NamespaceStore nsStore;
+  private final NamespaceQueryAdmin namespaceQueryAdmin;
 
   @Inject
-  NamespaceExistenceVerifier(NamespaceStore nsStore) {
-    this.nsStore = nsStore;
+  NamespaceExistenceVerifier(NamespaceQueryAdmin namespaceQueryAdmin) {
+    this.namespaceQueryAdmin = namespaceQueryAdmin;
   }
 
 
   @Override
   public void ensureExists(NamespaceId namespaceId) throws NotFoundException {
     if (!NamespaceId.SYSTEM.equals(namespaceId)) {
-      if (nsStore.get(namespaceId.toId()) == null) {
+      boolean exists = false;
+      try {
+        exists = namespaceQueryAdmin.exists(namespaceId.toId());
+      } catch (Exception e) {
+        Throwables.propagate(e);
+      }
+      if (!exists) {
```

Comment:
According to the contract of  this method if namespace does not exists we throw NotFoundExeption. How will that work ?